### ISSUE TITLE
Request for True Momentum Oscillator (TMO)

### ISSUE THREAD

**Issue by azimgiant** [twopirllc/pandas-ta#700]
First of all thank you for creating this project. It has been a great help. I wanted to request the TMO indicator as it has helped me a lot in my own personal trading. I just started to learn python to backtest my trading strategies, so my knowledge is at a beginner level. I tried using chatgpt to convert the pinescript to python however that wasn't much help. If this could be added in that would be awesome. Thanks!

```c
[https://www.tradingview.com/script/o9BQyaA4-True-Momentum-Oscillator/] 
Pinescript:
// @version=4
// This source code is subject to the terms of the Mozilla Public License 2.0 at https://mozilla.org/MPL/2.0/

// True Momentum Oscillator
// by SparkyFlary

study(title="True Momentum Oscillator", overlay=false)
    
length = input(14, title="Length")
calcLength = input(5, title="Calc length")
smoothLength = input(3, title="Smooth length")
lengthType = input("EMA", title="Length moving average type", options=["EMA", "SMA", "RMA"])
calcLengthType = input("EMA", title="Calc length moving average type", options=["EMA", "SMA", "RMA"])
smoothLengthType = input("EMA", title="Smooth length moving average type", options=["EMA", "SMA", "RMA"])

//function for choosing moving averages
f_ma(type, src, len) =>
    float result = 0
    if type == "EMA"
        result := ema(src, len)
    if type == "SMA"
        result := sma(src, len)
    if type == "RMA"
        result := rma(src, len)
    result
    
o = open
c = close
s = 0
for i = 0 to length
    s := s + (c > o[i] ? 1 : c < o[i] ? - 1 : 0)
data = s
    
MA = f_ma(lengthType, data, calcLength)
Main = f_ma(calcLengthType, MA, smoothLength)
Signal = f_ma(smoothLengthType, Main, smoothLength)

ob = hline(round(length * .7), title="overbought cutoff line", color=color.gray, linestyle=hline.style_solid)
os = hline(-round(length * .7), title="oversold cutoff line", color=color.gray, linestyle=hline.style_solid)
upper = hline(length, title="upper line", color=color.red, linestyle=hline.style_solid)
lower = hline(-length, title="lower line", color=color.green, linestyle=hline.style_solid)
zero = hline(0, title="zero line", color=color.gray, linestyle=hline.style_solid)

mainPlot = plot(Main, title="main line", color=Main>Signal?color.green:color.red, linewidth=2)
signalPlot = plot(Signal, title="signal line", color=Main>Signal?color.green:color.red)
crossPlot = plot(cross(Main,Signal)?Main:na, title="crossover dot", color=Main>Signal?color.green:color.red, style=plot.style_circles, linewidth=3)

fill(mainPlot, signalPlot, title="main and signal area", color=Main>Signal?color.green:color.red)
fill(ob, upper, title="overbought zone", color=color.red)
fill(os, lower, title="oversold zone", color=color.green)
```

**Comment by luisbarrancos** [comment]
@twopirllc I actually have an implementation which i didn't submit due to not knowing exactly which definition to use. Do we have a clear TMO definition from an authoritative source, or do we assume the TMO definition in, for example, [Trading View TMO addon](https://www.tradingview.com/script/VRwDppqd-True-Momentum-Oscillator/) is the one to be implemented?

The implementation from TV is the one i added and plotted with PlotLy as you can see here
![newplot](https://github.com/twopirllc/pandas-ta/assets/387352/22520c51-c391-47c7-b9f1-6ce0391b819a)

**Comment by luisbarrancos** [comment]
@twopirllc feel free to assign this to me.

**Comment by twopirllc** [comment]
@luisbarrancos 

Lol. No need to wait for my assignment. Submit a PR whenever you like. 😎 

KJ

**Comment by azimgiant** [comment]
@luisbarrancos Thanks for taking a look into this! The Trading View definition is actually the most accurate I believe, the one you linked. You could use that one however, I personally do not find the histogram or the smooth line to of any help. I think the main and smooth momentum lines are the most integral part of the indicator. You could still add them as an option.

**Comment by luisbarrancos** [comment]
@azimgiant no problem. There are several competing definitions in TV actually, with one at least having different default periods for the EMAs. You can find the details in the documentation of the indicator in the pull request [here](https://github.com/twopirllc/pandas-ta/pull/729)
By default it only outputs the main TMO signal and the smoothed TMO signal. Their respective momentums are also returned if you choose to enable that option. Building the histogram is trivial with the TMO main signal and smoothed signal. In that PR you'll find a jupyterlab notebook that shows you how to add traces for the main lines, the momentum, and how to build the shaded areas and shaded histograms from the returned indicator information.

In short: the functionality is the baseline common one provided in all TV indicators, with optional momentums provided by some indicators. The choice of moving average is **EMA** by default, but you can use **any** of the Pandas-TA moving averages as well - not only the ones that are provided in TV.

**Comment by azimgiant** [comment]
@luisbarrancos Thanks for implementing the indicator to this library! Appreciate that!!

I was trying your example on the TMO indicator, however, ran into this error:  ImportError: cannot import name 'tmo' from 'pandas_ta.momentum'. Has the indicator not yet been integrated to pandas-ta yet?

**Comment by luisbarrancos** [comment]
@azimgiant  the [pull request is open](https://github.com/twopirllc/pandas-ta/pull/729) and it wasn't merged into the *development* branch yet, so you need to [checkout the PR yourself](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally), and build pandas-ta in your environment or wait until it's merged into the development branch.

**Comment by luisbarrancos** [comment]
@azimgiant @twopirllc already merged the PR into the development branch. @twopirllc i think this issue can be closed.

**Comment by twopirllc** [comment]
@luisbarrancos,

Ahh yes! I forgot all about this request. Thanks for reminding me.

Thanks again for adding it to library. 😎 

KJ
